magento/magento2
# fixed negative  children_count after deleting categories

### Description (*)


### Related Pull Requests


### Fixed Issues (if relevant)

1. magento/magento2#27969: children_count can become negative after deleting categories

### Manual testing scenarios (*)

1.Log in into admin backend
2.Navigate to Catalog -> Categories
3.Click "Add Subcategory"
4.Fill category name with "Parent 1", Save
5.Click "Add Subcategory"
6.Fill category name with "Child 1", Save
7.Click "Add Subcategory"
8.Fill category name with "Child 2", Save
9.Click "Add Subcategory"
10.Fill category name with "Child 3", Save. At this point this category should have level = 5
11.Click on "Child 1" category
12.Delete category

### Questions or comments


### Contribution checklist (*)
